figma-shop/src/app/pages/products/products.component.ts

```diff
@@ -11,6 +11,8 @@ import { ProductServiceService } from '../../shared/services/product-service.ser
 
 export class ProductsComponent implements OnInit {
   public allProducts: Array<ProductInterface>;
+  public productsViewed: Array<ProductInterface> = [];
```

Comment:
Better use type definitions for arrays like this:
```suggestion
  public productsViewed: ProductInterface[] = [];
```
But you can leave as it is